packages/redux-subspace/src/enhancers/applySubspaceMiddleware.js

```diff
@@ -12,14 +12,28 @@ const applySubspaceMiddleware = (...middlewares) => (createSubspace) => (store)
 
     const subspacedStore = createSubspace(store)
 
-    const chain = middlewares.map((middleware) => middleware(subspacedStore))
+    let getState = subspacedStore.getState
+    let dispatch = subspacedStore.dispatch
+
+    const { namespace, rootStore, subspaceTypes, processAction } = subspacedStore
+
+    const middlewareApi = {
+        getState: (...args) => getState(...args),
+        dispatch: (...args) => dispatch(...args),
```

Comment:
This is now much more similar to [Redux's own implementation](https://github.com/reactjs/redux/blob/master/src/applyMiddleware.js#L27).